src/commonMain/kotlin/OpenTest4kH.kt

```diff
@@ -120,6 +120,26 @@ expect val ValueWrapper.identityHashCode: Int
  */
 expect val ValueWrapper.ephemeralValue: Any?
 
+/**
+ * Constructs an `AssertionFailedError` with an optional message and cause
+ * but without expected/actual values.
+ *
+ * @param message the detail message; `null` or blank will be converted to the empty `String`
+ * @param cause the cause of the failure
+ */
+inline fun assertionFailedError(message: String? = null, cause: Throwable? = null) = AssertionFailedError(message, cause)
+
+/**
+ * Constructs an `AssertionFailedError` with a message,
+ * expected/actual values, and optional cause.
+ *
+ * @param message the detail message; `null` or blank will be converted to the empty `String`
+ * @param expected the expected value; may be `null`
+ * @param actual the actual value; may be `null`
+ * @param cause the cause of the failure
+ */
+inline fun assertionFailedError(message: String? = null, expected: Any? = null, actual: Any? = null, cause: Throwable? = null) = AssertionFailedError(message, expected, actual, cause)
```

Comment:
To make a transition easier I've added constructor functions with default arguments. The only thing I couldn't do is name it `AssertionFailedError` as it's not compilable (names clash). Hence the first letter is lowercased. All dependent code which used default arguments will have to lowercase first letter to use this version.

Reply:
I've tried the constructions like
```
constructor(message: String?) : this(message, null)
```
, but get something like `Explicit delegation call for constructor of an expected class is not allowed`. Your example is not an `expect class`. You should play with it yourself to doublecheck if I missed something.

Reply:
I think you can leave off the delegation, it should just forward to the overloads defined in the actual right?

Reply:
Like this:

```kotlin
expect open class AssertionFailedError : AssertionError {

    /**
     * Constructs an `AssertionFailedError` with an optional message and cause
     * but without expected/actual values.
     */
    constructor()

    /**
     * Constructs an `AssertionFailedError` with an optional message and cause
     * but without expected/actual values.
     *
     * @param message the detail message; `null` or blank will be
     */
    constructor(message: String?)
```

```kotlin
actual open class AssertionFailedError : AssertionError {

    private val _cause: Throwable?
    val expected: ValueWrapper?
    val actual: ValueWrapper?

    actual constructor(): this(null)

    actual constructor(message: String?): this(message, null)
```

Reply:
It actually might work. But what amount of constructors do you want to see there? All of them might look ugly and non-kotlin way:

```kotlin
constructor(message: String?, cause: Throwable?)
constructor()    
constructor(message: String?)
constructor(cause: Throwable?)
constructor(message: String?, cause: Throwable?)

constructor(message: String?, expected: Any?, actual: Any?, cause: Throwable?)
constructor(message: String?, expected: Any?, actual: Any?)
constructor(expected: Any?, actual: Any?, cause: Throwable?)
constructor(expected: Any?, actual: Any?)
```
There is a tradeoff between good design and backwatd compatability.

Reply:
BTW, just came to my mind... What do you think about this alternative approach (Draft):

```kotlin
inline fun AssertionFailedError(message: String? = null, cause: Throwable? = null) = AssertionFailedError(message, cause)
inline fun AssertionFailedError(message: String? = null, expected: Any? = null, actual: Any? = null) = AssertionFailedError(message, expected, actual,null)
inline fun AssertionFailedError(expected: Any? = null, actual: Any? = null, cause: Throwable? = null) = AssertionFailedError(null, expected, actual,cause)


expect open class AssertionFailedError(message: String?, expected: Any?, actual: Any?, cause: Throwable?) : AssertionError 
```

Reply:
I've reworked the constructors. Please, check the latest version.